### ISSUE TITLE
Wikilinks and Date options

### ISSUE THREAD

**Issue by pwntester** [chrisgurney/things2md#3]
Hi,

First of all, thanks for writing this, I was about to implement it myself and found your project does 95% of what I was trying to achieve.

The only two options I miss are:
- add surrounding `[[]]` to project names so they get linked in Obsidian. (I use the same project names in Things and Obsidian so there will always be a match).
- a `--date` option to query for a specific date in the past.

Are these features you would merge? If so, I can work on them and send a PR, otherwise, no problem, will maintain them in a separate fork.

Thanks!

**Comment by chrisgurney** [comment]
Thank you for your comment, and for your offer! Both would make great additions.

I'd gladly review a PR for each addition:

1. Linking project names: To keep this generic and not Obsidian specific, I'm thinking the approach I'd like to take is have the `--format` argument accept `obsidian` as an option. (I just changed it to accept a list, so this is now a possibility.) If present, then project names would be wrapped with `[[]]`.
    - I also keep my project names in sync in Obsidian, with the exception that I strip off emojis I have in Things. I'm open to making this an option.
    - As an aside, take a look at the output from the `--oprojects` argument, which I use to just export a list of projects; each also includes links back to the projects in Things. After you've made any changes I'll probably refactor this to not be an Obsidian-specific argument, if that makes sense.
 2. A `--date` argument makes perfect sense, and something I wanted personally, to back-fill daily notes. Let's keep it to the ISO date format: `YYYY-MM-DD`.

**Comment by chrisgurney** [comment]
I've got a patch in the works for wikilinks on project names. Instead of `--format obsidian` changing it to `--format wikilinks` to make it even more generic.

**Comment by chrisgurney** [comment]
@pwntester Wikilinks (for project names) are now live!

`python3 things2md.py --today --format wikilinks`

Looking to see if I can add the `--date` argument without too much work.

**Comment by chrisgurney** [comment]
@pwntester I've added preliminary support for `--date` but note this caveat around (I think) time zones: https://github.com/chrisgurney/things2md/pull/2#issuecomment-1962999878

**Comment by chrisgurney** [comment]
@pwntester Update: I think I have a fix in place (given my workaround [noted previously](https://github.com/chrisgurney/things2md/pull/2#issuecomment-1963016916)) but it's not great. Following up with the maintainer to see if there's a more elegant solution, but for now it should work for you. LMK how it goes!

**Comment by chrisgurney** [comment]
@pwntester Your suggestions are working beautifully in my Daily Notes. Thanks again!

![CleanShot 2024-02-25 at 22 41 26](https://github.com/chrisgurney/things2md/assets/17337703/01631ad9-a3fb-4a7f-8d89-0d0a06bab086)

**Comment by chrisgurney** [comment]
Update: I've published [gitactivity2md](https://github.com/chrisgurney/gitactivity2md/) and it now supports a `--date` argument!

@pwntester One more thing that may interest you: I have a separate Python script almost ready to share that dumps a list of your Git commits and PRs in markdown format, from a given list of repos. 

My use case is to keep a record of my Git activity in my Daily Notes (rather than relying on having written tasks or notes about that activity).

Here's a sense of what the output currently looks like:

```
% python3 github2md.py --repos things2md --range "1 day ago"

- things2md
	- doc: updated --date beta comment • 2024-02-25
	- fix: --date argument gets all tasks completed on given date • 2024-02-25
	- refactor: perf improvement for format arg • 2024-02-25
	- feat: added --date argument (beta) • 2024-02-25
	- docs: correction about how import works • 2024-02-25
	- docs: removed --today beta tag and added note about evening tasks • 2024-02-25
	- docs: updated format argument • 2024-02-25
	- feat: format arg now supports noemojis and wikilinks • 2024-02-25
	- feat: format arg now supports noemojis and wikilinks • 2024-02-25
```

**Comment by pwntester** [comment]
Awesome! Thanks bery much, Im thinking on commiting my things3 db to my obsidian repo and schedule a github action to dump the logbook of the day everyday at 3:00 am or so. Your git activity script sounds like a great addition to that

**Comment by chrisgurney** [comment]
@pwntester Let me know if you start testing it out. I'd love feedback, as I'm sure it'll selfishly make it better for me as well. 😄

**Comment by chrisgurney** [comment]
@pwntester FYI I've just released v2.0 of `things2md` which supports output templates, and more granular options via a configuration file:

https://github.com/chrisgurney/things2md

If you do want to try this out, make sure you see the release notes as configuration and templates removes the need for several existing parameters.

https://github.com/chrisgurney/things2md/releases/tag/v2.0

Let me know if you have any questions!
